engine/src/main/java/org/terasology/rendering/nui/layers/mainMenu/JoinGameScreen.java

```diff
@@ -387,4 +407,19 @@ private String getModulesText(Future<ServerInfoMessage> info) {
         }
     }
 
+    public boolean onKeyEvent(NUIKeyEvent event) {
+        if (event.isDown() && event.getKey() == Keyboard.Key.R) {
+            refresh();
+            }
+            return false;
+        }
+
+    public void refresh() {
+        logger.info("Refresh");
```

Comment:
I'd either remove this message or expand on it a bit (i.e. "Refreshing info for server X").